SSA/Experimental/Bits/Fast/BitStream.lean

```diff
@@ -373,12 +373,41 @@ variable {w : Nat} {x y : BitVec w} {a b a' b' : BitStream}
 
 local infix:20 " ≈ʷ " => EqualUpTo w
 
--- TODO: These sorries are difficult, and will be proven in a later Pull Request.
+-- TODO: This sorry is difficult, and will be proven in a later Pull Request.
 theorem ofBitVec_sub : ofBitVec (x - y) ≈ʷ (ofBitVec x) - (ofBitVec y)  := by
   sorry
 
+theorem Nat.two_mod_mod (x : Nat) (w : Nat) (_ : 0 < w) : x % 2 ^ w % 2 = x % 2 := by
```

Comment:
```suggestion
theorem Nat.mod_two_pow_mod_two (x : Nat) (w : Nat) (_ : 0 < w) : x % 2 ^ w % 2 = x % 2 := by
```

This, and the other `Nat` lemmas, should be moved to a separate file, maybe `SSA/Experimental/Bits/ForLean.lean` to mark them as candidates for upstreaming